code/modules/orbit/orbit.dm

```diff
@@ -0,0 +1,103 @@
+/datum/orbit
+	var/atom/movable/orbiter
+	var/atom/orbiting
+	var/lock = TRUE
+	var/turf/lastloc
+
+/datum/orbit/New(_orbiter, _orbiting, _lock)
+	orbiter = _orbiter
+	orbiting = _orbiting
+	OnMove()
+	lock = _lock
+
+	//at this point we could get deleted and orbiter/orbiting could be null.
+	if (orbiting)
+		if (!orbiting.orbiters)
+			orbiting.orbiters = list()
+		orbiting.orbiters += src
+
+	if (orbiter)
+		if (orbiter.orbiting)
+			orbiter.stop_orbit()

```

Comment:
@AdamElTablawy This line of code stops the existing orbit when you start up a new orbit